tools/manifest/sourcefile.py

```diff
@@ -62,6 +66,21 @@ def read_script_metadata(f: BinaryIO, regexp: Pattern[bytes]) -> Iterable[Tuple[
 
         yield (m.groups()[0].decode("utf8"), m.groups()[1].decode("utf8"))
 
+def read_spec_link(f: BinaryIO, regexp: Pattern[bytes]) -> Iterable[Text]:
+    """
+    Yields spec information from the file-like object `f`,
+    as specified according to a supplied regexp.
+    `regexp` - Regexp containing one group, a href string.
+    """
+    for line in f:
+        assert isinstance(line, bytes), line
+        m = regexp.match(line)
```

Comment:
Please don't use a regex here; this should have something more like `timeout_nodes` via actually parsing the HTML. We should only fall back to anything else when we're not HTML/XML, should probably just use read_script_metadata (or maybe via some indirection) then.

Reply:
After addressing your comments, I realized that there is already implementation to extract spec information, `spec_links`. I will add the tests instead